Vostok.ClusterClient.Topology.SD/Transforms/NeverForgetReplicasTransform.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using Vostok.Commons.Helpers.Topology;
+using Vostok.ServiceDiscovery.Abstractions;
+
+namespace Vostok.Clusterclient.Topology.SD.Transforms
+{
+    [PublicAPI]
+    public class NeverForgetReplicasTransform : IServiceTopologyTransform
+    {
+        private readonly HashSet<Uri> detectedReplicas = new HashSet<Uri>(ReplicaComparer.Instance);
```

Comment:
Is it even safe to use `ReplicaComparer` here? I mean, it totally disregards the path portion of the `Uri`, which might be acceptable for Snitch where this comparer once originated, but not correct in general.